src/platform-aws.cpp

```diff
@@ -1082,3 +1083,13 @@ void PlatformAWS::sort_rails(struct fi_info **info_list, size_t num_rails, size_
 
 	return;
 }
+
+bool PlatformAWS::is_ec2_instance() {
+	const std::string& platform_override = ofi_nccl_platform.get();
```

Comment:
We might want to add some checks that would make sure that the `ofi_nccl_platform` value set in the loaded network library and loaded tuner library are the same so that both libraries are configured correctly. E.g. maybe checking that  `ofi_nccl_platform.get_source() != ParamSource::API` to validate that that `ofi_nccl_platform` wasn't set in the code through a `ofi_nccl_platform.set()` call that could have only happened in either the net library or the tuner library and not the other.